components/01-atoms/BlockExplorerExternalLinkButton.tsx

```diff
@@ -0,0 +1,38 @@
+import { ExternalLinkIcon } from "@/components/01-atoms";
+import { EthereumAddress } from "@/lib/shared/types";
+import { useNetwork } from "wagmi";
+
+export const BlockExplorerExternalLinkButton = ({
+  address,
+  label,
+}: {
+  address: EthereumAddress;
+  label?: string;
+}) => {
+  const { chain } = useNetwork();
+
+  if (!address) return null;
+
+  const displayEllipsedAddress = address.getEllipsedAddress();
+
+  const blockExplorer = `${
+    chain?.blockExplorers?.default.url
+  }/address/${address?.toString()}`;
```

Comment:
After the conditional above (`if (!address) return null;`) is this question mark still needed?

```suggestion
  }/address/${address.toString()}`;
```